worker/uniter/runner/jujuc/jujuctesting/unit.go

```diff
@@ -57,3 +60,12 @@ func (c *ContextUnit) SetPodSpec(specYaml string) error {
 	c.info.ContainerSpec = specYaml
 	return nil
 }
+
+func (c *ContextUnit) CloudSpec() (*params.CloudSpec, error) {
+	c.stub.AddCall("CloudSpec")
+	if err := c.stub.NextErr(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	c.info.CloudSpec = params.CloudSpec{}
+	return &c.info.CloudSpec, nil
```

Comment:
we may want to investigate gomock to allow us to auto-generate all of this file, though its probably not worth the learning curve to do so *today* it likely makes it much easier to maintain later.